### PR TITLE
Fixing the problem of configuring an sftp bucket with the wizard option

### PR DESCRIPTION
**Pull request by lucasncabral**
Was observed that an error was being returned ('NoneType' object is not subscriptable) when trying to configure an sftp bucket using the `--wizard-config` option of the create command. This PR fixes this issue. In addition, a refactoring was performed on the wizard methods to decouple code and structure user input.

**Observed behavior:**
```
$ ml-git models create models-ex --categories=test --mutability=strict --wizard-config
_ Current configured storages _
   
1 - s3h - test-bucket
X - New Data Storage
   
_Which storage do you want to use (a number or new data storage)? _ x
Please specify the storage type ['s3', 's3h', 'azureblobh', 'gdriveh', 'gdrive', 'sftph']: _ sftph
Please specify the bucket name: _ sftph
Please specify the git repository for ml-git models metadata: _ sftph-repository
INFO - Admin: Add storage [sftph://sftph]
ERROR - MLGit: 'NoneType' object is not subscriptable
A complete log of this run can be found in: ml-git_execution.log
```

**Behavior after adjustment:**
```
$ ml-git models create models-ex --categories=test --mutability=strict --wizard-config
Current configured storages:

[1] - s3h - test-bucket
[X] - Create new data storage

Inform the storage do you want to use: x
Inform the storage type ['s3h', 'azureblobh', 'gdriveh', 'sftph']: sftph
Inform the bucket name: sftp_dir
Inform the endpoint URL: <endpoint-url>
Inform the username: lucasncabral
Inform the credentials path: <path-to-credentials>
Inform the port: 9000
INFO - Admin: Add storage [sftph://sftp_dir]
INFO - MLGit: Model artifact created.
```

### CODE REVIEW

File: ml_git/config.py [hunk 1]
```diff
@@ -55,6 +56,8 @@
 
 }
 
+USER_INPUT_MESSAGE = 'Inform the {}: > '
```
Comment (gilMars) [inline ml_git/config.py:59]: Remove the angular brackets

File: ml_git/config.py [hunk 2]
```diff
@@ -344,51 +347,71 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
...
-    endpoint = None
-    git_repo = None
+def _configure_metadata_remote(repo_type):
     config = mlgit_config_load()
     try:
-        int(selected)  # the user select one storage from the list
-        has_new_storage = False
-        # extract necessary info from the storage in spec
+        git_repo = config[repo_type]['git']
+        if git_repo == '':
```
Comment (gilMars) [inline ml_git/config.py:354]: Change  `if git_repo == ''` to `if not git_repo`

File: ml_git/config.py [hunk 3]
```diff
@@ -344,51 +347,71 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
...
+    storage_type = input(USER_INPUT_MESSAGE.format('storage type {}'.format(str(storages_types)))).lower()
+
+    credential_profile = None
+    endpoint = None
+    sftp_configs = None
+
+    if storage_type not in storages_types:
+        raise RuntimeError(output_messages['ERROR_INVALID_STORAGE_TYPE'])
+    bucket = input(USER_INPUT_MESSAGE.format('bucket name'))
+    if storage_type in (StorageType.S3.value, StorageType.S3H.value):
```
Comment (gilMars) [inline ml_git/config.py:373]: If the user type 's3' as storage type, the first condition is satisfied throwing the RuntimeError. Is it the correct behavior?

File: ml_git/config.py [hunk 4]
```diff
@@ -344,51 +347,71 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
...
+        credential_profile = input(USER_INPUT_MESSAGE.format('credentials'))
+        endpoint = input('If you are using MinIO inform the endpoint URL, otherwise press ENTER: > ')
```
Comment (gilMars) [inline ml_git/config.py:375]: Remove angular brackets.

File: ml_git/config.py [hunk 5]
```diff
@@ -344,51 +347,71 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
...
+    print('Current configured storages:\n   ')
+    configured_storages = config_load()[STORAGE_CONFIG_KEY]
+
+    valid_buckets, temp_map = _get_configured_buckets(configured_storages)
+    print('[X] - Create new data storage\n   ')
+    selected = input(USER_INPUT_MESSAGE.format('storage do you want to use'))
+
+    if selected.upper() == 'X':
+        storage_type, bucket = _create_new_bucket()
+    elif selected.isnumeric() and int(selected) in range(1, len(valid_buckets) + 1):
```
Comment (gilMars) [inline ml_git/config.py:409]: Extract the condition, `int(selected) in range(1, len(valid_buckets) + 1)` to a variable to the improve readability.

File: ml_git/config.py [hunk 6]
```diff
@@ -344,51 +347,71 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
...
-    git_repo = None
+def _configure_metadata_remote(repo_type):
     config = mlgit_config_load()
     try:
-        int(selected)  # the user select one storage from the list
-        has_new_storage = False
-        # extract necessary info from the storage in spec
+        git_repo = config[repo_type]['git']
+        if git_repo == '':
+            raise Exception('Need configure a remote repository.')
```
Comment (GustavoMaciel) [inline ml_git/config.py:355]:
```suggestion
        if not config[repo_type]['git']:
            raise Exception('Need to configure a remote repository.')
```

File: ml_git/config.py (hunk 5 above)
Comment (GustavoMaciel) [reply ml_git/config.py:409]: It could result in a parse exception, perhaps saving the range(...) in a variable would be enough to improve readability

Comment (gilMars) [reply ml_git/config.py:409]: I agree with @GustavoMaciel's  argument.

File: ml_git/config.py (hunk 1 above)
Comment (lucasncabral) [reply ml_git/config.py]: The character has been removed from this and other necessary messages in 30d3b6b.

File: ml_git/config.py (hunk 2 above)
Comment (lucasncabral) [reply ml_git/config.py]: Change applied in 30d3b6b. Thanks.

File: ml_git/config.py (hunk 3 above)
Comment (lucasncabral) [reply ml_git/config.py]: The behavior is correct. This condition has been updated to take into account only the `S3H`.

File: ml_git/config.py (hunk 5 above)
Comment (lucasncabral) [reply ml_git/config.py]: The conditional was updated in b1a3ec3 to use a variable for the range. Final result:

```
elif selected.isnumeric() and int(selected) in valid_buckets_options:
```

File: ml_git/config.py (hunk 6 above)
Comment (lucasncabral) [reply ml_git/config.py]: Suggestion applied in 30d3b6b.

**Review by GustavoMaciel — APPROVED** [review]

**Review by gilMars — APPROVED** [review]

**Review by andersonpablosilva — APPROVED** [review]

**Pull request merged**